Fix failure on `test_unshun_algorithm-t.cpp`

## Description

During a CI run `test_unshun_algorithm-t.cpp` failed, the logs suggest that the test attempted to recover a `SHUNNED` server in an interval that might be less than the configured `mysql-shun_recovery_time_sec`:

`test_unshun_algorithm-t.cpp`:
```
...
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.012278  running query: DO /* ;hostgroup=0 */ 1
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.012326  running query: SELECT status,hostgroup_id FROM runtime_mysql_servers WHERE port=3300 order by hostgroup_id
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017673  Status found for server '0:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017706  Status found for server '10:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017714  Status found for server '11:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017720  Status found for server '12:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017726  Status found for server '13:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017733  Status found for server '14:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017739  Status found for server '15:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017746  Status found for server '16:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017753  Status found for server '17:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017759  Status found for server '18:127.0.0.1:3300' was 'ONLINE'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: # 2025-04-28 17:13:50.017766  Status found for server '19:127.0.0.1:3300' was 'SHUNNED'
[2025-04-28 17:13:50] DEBUG    [proxysql-tester.py:714] msg: not ok 26 - 2025-04-28 17:13:50.017775 - Server from first hg was set 'ONLINE' while others remained 'ONLINE'
```

`proxysql.log`:
```
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3300 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3301 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3302 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3303 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3304 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3305 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3306 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3307 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3308 1234
2025-04-28 17:13:46 Admin_Handler.cpp:2568:admin_session_handler(): [WARNING] Received PROXYSQL_SIMULATOR command: PROXYSQL_SIMULATOR mysql_error 19 127.0.0.1:3309 1234
2025-04-28 17:13:46 MyHGC.cpp:186:get_random_MySrvC(): [ERROR] Hostgroup 0 has no servers available! Checking servers shunned for more than 1 second
2025-04-28 17:13:46 ProxySQL_Cluster.cpp:637:set_checksums(): [INFO] Cluster: detected a new checksum for mysql_variables from peer 127.0.0.1:6032, version 874, epoch 1745860425, checksum 0x826950CA5BFFF3CD . Not syncing yet ...
2025-04-28 17:13:46 ProxySQL_Cluster.cpp:643:set_checksums(): [INFO] Cluster: checksum for mysql_variables from peer 127.0.0.1:6032 matches with local checksum 0x826950CA5BFFF3CD , we won't sync.
2025-04-28 17:13:46 PgSQL_Monitor.cpp:642:handle_async_connect_cont(): [ERROR] Monitor connect failed   addr='127.0.0.1:15432' error='FATAL:  password authentication failed for user "monitor"'
2025-04-28 17:13:46 PgSQL_Monitor.cpp:642:handle_async_connect_cont(): [ERROR] Monitor connect failed   addr='127.0.0.1:15432' error='FATAL:  password authentication failed for user "monitor"'
2025-04-28 17:13:47 MyHGC.cpp:98:get_random_MySrvC(): [INFO] Unshunning server 127.0.0.1:3300.
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2389:unshun_server_all_hostgroups(): [INFO] Calling unshun_server_all_hostgroups() for server 127.0.0.1:3300 . Arguments: 1745860427 , 1 , 0
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 10:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 11:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 18:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 17:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 12:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 13:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 14:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 15:127.0.0.1:3300 . time_last_detected_error=1745860425
2025-04-28 17:13:47 MySQL_HostGroups_Manager.cpp:2417:unshun_server_all_hostgroups(): [INFO] Unshunning server 16:127.0.0.1:3300 . time_last_detected_error=1745860425
```

As can be seeing in the logs, the latest server to be shunned `19:127.0.0.1:3300` may have been shunned within the same second of the un-shunning attempt (due to actual time diff or second rounding error). To prevent this, the test should wait at least `mysql-shun_recovery_time_sec + 1` after attempting to recover a server from `SHUNNED`.

--- a/test/tap/tests/test_unshun_algorithm-t.cpp
+++ b/test/tap/tests/test_unshun_algorithm-t.cpp
@@ -289,12 +289,9 @@ int configure_mysql_shunning_variables(MYSQL* proxysql_admin) {
 
 int test_unshun_algorithm_behavior(MYSQL* proxysql_mysql, MYSQL* proxysql_admin) {
 	// Configure Admin variables with lower thresholds
-	MYSQL_QUERY(proxysql_admin, "SET mysql-shun_recovery_time_sec=1");
-	diag("Line:%d running admin query: SET mysql-shun_recovery_time_sec=1", __LINE__);
-
+	MYSQL_QUERY(proxysql_admin, ("SET mysql-shun_recovery_time_sec=" + _TO_S(SHUN_RECOVERY_TIME)).c_str());
 	// Set verbosity up for extra information in ProxySQL log
 	MYSQL_QUERY(proxysql_admin, "SET mysql-hostgroup_manager_verbose=3");
-	diag("Line:%d running admin query: SET mysql-hostgroup_manager_verbose=3", __LINE__);
 
 	// Configure the relevant variables for the desired UNSHUNNING behavior
 	if (configure_mysql_shunning_variables(proxysql_admin)) {
@@ -333,6 +330,9 @@ int test_unshun_algorithm_behavior(MYSQL* proxysql_mysql, MYSQL* proxysql_admin)
 		int shunn_err = shunn_all_servers(proxysql_admin);
 		if (shunn_err) { return EXIT_FAILURE; }
 
+		diag("Sleeping SHUN_RECOVERY_TIME(%d+1)s before server wakeup...", SHUN_RECOVERY_TIME);
+		sleep(SHUN_RECOVERY_TIME + 1);
+
 		for (uint32_t i = 0; i < SERVERS_COUNT; i++) {
 			wakup_target_server(proxysql_mysql, i);
 
@@ -352,6 +352,10 @@ int test_unshun_algorithm_behavior(MYSQL* proxysql_mysql, MYSQL* proxysql_admin)
 
 		int shunn_err = shunn_all_servers(proxysql_admin);
 		if (shunn_err) { return EXIT_FAILURE; }
+
+		diag("Sleeping SHUN_RECOVERY_TIME(%d+1)s before server wakeup...", SHUN_RECOVERY_TIME);
+		sleep(SHUN_RECOVERY_TIME + 1);
+
 		diag(" "); // empty line
 		for (uint32_t i = 0; i < SERVERS_COUNT; i++) {
 			wakup_target_server(proxysql_mysql, i);
@@ -372,6 +376,9 @@ int test_unshun_algorithm_behavior(MYSQL* proxysql_mysql, MYSQL* proxysql_admin)
 
 		int shunn_err = shunn_all_servers(proxysql_admin);
 		if (shunn_err) { return EXIT_FAILURE; }
+
+		diag("Sleeping SHUN_RECOVERY_TIME(%d+1)s before server wakeup...", SHUN_RECOVERY_TIME);
+		sleep(SHUN_RECOVERY_TIME + 1);
 
 		for (uint32_t i = 0; i < SERVERS_COUNT; i++) {
 			wakup_target_server(proxysql_mysql, i);

